boa_engine/src/builtins/intl/date_time_format.rs

```diff
@@ -0,0 +1,114 @@
+//! This module implements the global `Intl.DateTimeFormat` object.
+//!
+//! `Intl.DateTimeFormat` is a built-in object that has properties and methods for date and time i18n.
+//!
+//! More information:
+//!  - [ECMAScript reference][spec]
+//!
+//! [spec]: https://tc39.es/ecma402/#datetimeformat-objects
+
+use crate::{
+    builtins::BuiltIn,
+    context::intrinsics::StandardConstructors,
+    object::internal_methods::get_prototype_from_constructor,
+    object::{JsObject, ObjectData, ObjectInitializer},
+    property::Attribute,
+    symbol::WellKnownSymbols,
+    Context, JsResult, JsValue,
+};
+
+use boa_gc::{Finalize, Trace};
+use boa_profiler::Profiler;
+use tap::{Conv, Pipe};
+
+/// JavaScript `Intl.DateTimeFormat` object.
+#[derive(Debug, Clone, Trace, Finalize)]
+pub struct DateTimeFormat {
+    initialized_date_time_format: bool,
+    locale: String,
+    calendar: String,
+    numbering_system: String,
+    time_zone: String,
+    weekday: String,
+    era: String,
+    year: String,
+    month: String,
+    day: String,
+    day_period: String,
+    hour: String,
+    minute: String,
+    second: String,
+    fractional_second_digits: String,
+    time_zone_name: String,
+    hour_cycle: String,
+    pattern: String,
+    bound_format: String,
+}
+
+impl BuiltIn for DateTimeFormat {
+    const NAME: &'static str = "Intl.DateTimeFormat";
+
+    fn init(context: &mut Context) -> Option<JsValue> {
+        let _timer = Profiler::global().start_event(Self::NAME, "init");
+
+        let string_tag = WellKnownSymbols::to_string_tag();
+        ObjectInitializer::new(context)
+            .property(
+                string_tag,
+                Self::NAME,
+                Attribute::READONLY | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
+            )
+            .build()
+            .conv::<JsValue>()
+            .pipe(Some)
+    }
+}
+
+impl DateTimeFormat {
```

Comment:
```suggestion
impl DateTimeFormat {
    const NAME: &'static str = "DateTimeFormat";
    
    pub(super) fn init(context: &mut Context) -> JsObject {
        ...
    }
```

Since we are not creating a separate buitin object (this is a part of `Intl` object). We don't have to implement `BuiltIn` on it. Instead we create a static method `init` which returns the constructor object (`JsObject`). Using the `ConstructorBuilder`

Reply:
Done https://github.com/boa-dev/boa/pull/2025/files#diff-d3b068b8d1eb849b10fde7458b9cfc96c68353420893ef2ca613b258884c1756R47